index.d.ts

```diff
@@ -427,11 +427,12 @@ declare namespace Moleculer {
 	}
 
 	type ActionCacheEnabledFuncType = (ctx: Context<any, any>) => boolean;
+	type ActionCacheKeygenFuncType<P = Record<string, any>, M = any, K = any[]|null> = (actionName: string, params: P, meta: M, defaultKeys: K) => string;
 
-	interface ActionCacheOptions {
+	interface ActionCacheOptions<P = Record<string, any>, M = any, K = any[]|null> {
 		enabled?: boolean | ActionCacheEnabledFuncType;
 		ttl?: number;
-		keys?: Array<string>;
+		keys?: string[] | ActionCacheKeygenFuncType<P, M, K>;
```

Comment:
I must be missing something, but I don't see anywhere that `keys` can accept anything other than a `string[]`.  What code is checking this property for a function?  In the referenced PR I see that it allowed specification of a `keygen` property, but I'm not sure where you can override the value of `keys` with a function.

Reply:
Yeah, it came with #1004 pull as a new feature.

Reply:
@0x0a0d Shawn has right, the #1004 adds `keygen` property to the action `cache` definition and not change the `keys`.